Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/core/repository/FamilyTestRepository.kt

```diff
@@ -1,76 +1,125 @@
 package de.rki.coronawarnapp.familytest.core.repository
 
-import de.rki.coronawarnapp.coronatest.TestRegistrationRequest
+import de.rki.coronawarnapp.coronatest.qrcode.CoronaTestQRCode
 import de.rki.coronawarnapp.coronatest.type.TestIdentifier
 import de.rki.coronawarnapp.familytest.core.model.FamilyCoronaTest
-import de.rki.coronawarnapp.util.coroutine.AppScope
-import kotlinx.coroutines.CoroutineScope
+import de.rki.coronawarnapp.familytest.core.model.markBadgeAsViewed
+import de.rki.coronawarnapp.familytest.core.model.markDccCreated
+import de.rki.coronawarnapp.familytest.core.model.markViewed
+import de.rki.coronawarnapp.familytest.core.model.recycle
+import de.rki.coronawarnapp.familytest.core.model.restore
+import de.rki.coronawarnapp.familytest.core.model.updateResultNotification
+import de.rki.coronawarnapp.familytest.core.storage.FamilyTestStorage
+import de.rki.coronawarnapp.util.TimeStamper
 import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
 import javax.inject.Inject
+import javax.inject.Singleton
 
+@Singleton
 class FamilyTestRepository @Inject constructor(
-    @AppScope private val appScope: CoroutineScope
+    private val processor: BaseCoronaTestProcessor,
+    private val storage: FamilyTestStorage,
+    private val timeStamper: TimeStamper,
 ) {
 
-    val familyTests: Flow<Set<FamilyCoronaTest>> = flowOf(emptySet())
-    val recycledFamilyTests: Flow<Set<FamilyCoronaTest>> = flowOf(emptySet())
+    private val familyTestMap = storage.familyTestMap
+
+    val familyTests: Flow<Set<FamilyCoronaTest>> = storage.familyTestMap.map {
+        it.values.filter { !it.isRecycled }.toSet()
+    }
+
+    val familyTestRecyclingBin: Flow<Set<FamilyCoronaTest>> = storage.familyTestMap.map {
```

Comment:
Not necessary but maybe name it `familyTestRecycleBin` to be consistent with the naming throughout the app.